tests/contracts/ValidationRules.sol

```diff
@@ -22,65 +25,97 @@ contract Dummy {
     }
 }
 
-contract ValidationRulesStorage is IState {
-    IEntryPoint public entryPoint;
-    uint256 public state;
 
-    function funTSTORE() external override returns(uint256) {
-        assembly {
-            tstore(0, 1)
-        }
-        return 0;
-    }
+library ValidationRules {
+    using ValidationRules for string;
+    event Uint(uint);
 
-    function funSSTORE() external returns(uint256) {
-        assembly {
-            sstore(0, 1)
-        }
-        return 0;
+    function eq(string memory a, string memory b) internal pure returns (bool) {
+        return keccak256(bytes(a)) == keccak256(bytes(b));
     }
 
-
-    function funTLOAD() external returns(uint256) {
-        uint256 tval;
-        assembly {
-            tval := tload(0)
+    function startsWith(string memory str, string memory prefix) internal pure returns (bool) {
```

Comment:
it might worth using a standard strings library. they are more eficcient in code size, and avoid overbloat.

Reply:
It doesn't look like there is a library used widely enogh to be better than this code. It is only used for tests.
If you know about a 3rd party library with good string function set, please create a Jira issue, and I will migrate to it.